### ISSUE TITLE
SIMD: Implement 8-record mini-batching for forward-only networks

### ISSUE THREAD

**Issue by stservice** [stSoftwareAU/NEAT-TS#1209]
## Background

PR #1205 implemented 4-record SIMD mini-batching for forward-only networks, achieving a **31% speedup** by processing 4 records simultaneously using SIMD across records. This issue proposes extending that approach to 8-record batching for even greater performance.

## Proposal

Implement 8-record SIMD mini-batching by stacking two v128 operations to effectively process 8 records per neuron per iteration.

### Current Implementation (PR #1205)
```rust
// Gather activations from 4 different records at the same position
let acts = f32x4(act0[from], act1[from], act2[from], act3[from]);
// Broadcast weight to all 4 lanes
let weights = f32x4_splat(weight);
// FMA: acc = weights * acts + acc
acc = f32x4_relaxed_madd(weights, acts, acc);
```

### Proposed 8-Record Implementation
```rust
// First 4 records
let acts_0_3 = f32x4(act0[from], act1[from], act2[from], act3[from]);
// Second 4 records
let acts_4_7 = f32x4(act4[from], act5[from], act6[from], act7[from]);

let weights = f32x4_splat(weight);
acc_0_3 = f32x4_relaxed_madd(weights, acts_0_3, acc_0_3);
acc_4_7 = f32x4_relaxed_madd(weights, acts_4_7, acc_4_7);
```

## Expected Benefits

1. **Better CPU cache utilisation**: More records processed per neuron per iteration
2. **Amortised overhead**: SIMD setup overhead spread across more work
3. **Improved data locality**: Same activation weights shared across all 8 records
4. **Estimated speedup**: 50-70% over current 4-record implementation

## Implementation Notes

- Create `mse_sum_batch_8way()` internal helper
- Allocate 8 activation buffers instead of 4
- Fall back to 4-way for num_records < 8, scalar for < 4
- Apply to all loss functions: MSE, MAE, Cross-Entropy, MAPE, MSLE, Hinge

## Files to Modify

- `wasm_activation/src/lib.rs`: Add `weighted_sum_simd_8records()` and `mse_sum_batch_8way()`
- Update `mse_sum_batch_packed()` to route to 8-way when num_records >= 8

## Success Criteria

- Benchmark showing >40% improvement over 4-record batching for large datasets
- All existing tests pass with acceptable numerical precision
- Score precision remains within 0.001%

Related to: #1205, #1202
